### PR TITLE
UCSD Baseline Results Transformation: Update Receiving Facility (MSH-6)

### PR DESCRIPTION
**Pull request by basiliskus**
# Update Receiving Facility (MSH-6)

Implemented transformation rule: Update Receiving Facility (MSH-6) to value in ORC-21.10 and remove content from MSH-6.2 and MSH-6.3

## Issue

#1024

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)
- [x] I have added JavaDocs where required
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args)
+            throws RuleExecutionException {
+        try {
+            Bundle bundle = (Bundle) resource.getUnderlyingResource();
+            Organization receivingFacility = HapiHelper.getReceivingFacility(bundle);
+            Identifier namespaceIdentifier =
+                    HapiHelper.getHDNamespaceIdentifier(receivingFacility.getIdentifier());
+            if (namespaceIdentifier == null) {
```
Comment (saquino0827) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java:28]: For any kind of null checks, would it be better if we use FHIRPath and the `conditions` from [validation_definitions.json](https://github.com/CDCgov/trusted-intermediary/blob/main/etor/src/main/resources/validation_definitions.json#L6) as a way to ensure the values we need to manipulate are there?

**Comment by luis-pabon-tf** [comment]
I think this PR should base itself off of PR #1104 to make it easier to get both through.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        implements CustomFhirTransformation {
+
+    @Override
+    public void transform(FhirResource<?> resource, Map<String, String> args) {
+        Bundle bundle = (Bundle) resource.getUnderlyingResource();
+        DiagnosticReport diagnosticReport = HapiHelper.getDiagnosticReport(bundle);
+        if (diagnosticReport == null) {
+            return;
+        }
+        ServiceRequest serviceRequest = HapiHelper.getBasedOnServiceRequest(diagnosticReport);
```
Comment (jcrichlake) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java:25]: This method name is confusing, what are we getting based on the service request?

File: shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java [hunk 3]
```diff
@@ -121,6 +127,43 @@ public static MessageHeader.MessageDestinationComponent getMSH5MessageDestinatio
         return messageHeader.getDestinationFirstRep();
     }
 
+    // MSH-6 - Receiving Facility
+    public static Organization getMSH6Organization(Bundle bundle) {
```
Comment (jcrichlake) [inline shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java:131]: Do these all need to be static?

Comment (basiliskus) [reply shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java:131]: They are self contained and there's no need to instate them. Is there any reason not to use static?

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java (hunk 2 above)
Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java:25]: The `DiagnosticReport` has a `BasedOn` property that has a `ServiceRequest` in it. Do you have any suggestions how what to rename it to?

Comment (luis-pabon-tf) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java:25]: I think a name like `getServiceRequest` would do the job, and just have the parameter be of type `IBase`

Comment (basiliskus) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/UpdateReceivingFacilityWithOrderingFacilityIdentifier.java]: Renamed

**Review by tjohnson7021 — APPROVED** [review]

**Comment by basiliskus** [comment]
Verified that is working for end-to-end flow

**Pull request merged**